Source/WebKit/UIProcess/XR/openxr/OpenXRLayer.cpp

```diff
@@ -30,34 +40,83 @@ namespace WebKit {
 WTF_MAKE_TZONE_ALLOCATED_IMPL(OpenXRLayer);
 WTF_MAKE_TZONE_ALLOCATED_IMPL(OpenXRLayerProjection);
 
+OpenXRLayer::OpenXRLayer(UniqueRef<OpenXRSwapchain>&& swapchain, WebCore::GLContext& context)
+    : m_swapchain(WTFMove(swapchain))
+    , m_glContext(context)
+{
+}
+
+std::optional<PlatformXR::FrameData::ExternalTexture> OpenXRLayer::exportOpenXRTexture(PlatformGLObject openxrTexture)
+{
+    // Texture must be bound to be exported.
+    glBindTexture(GL_TEXTURE_2D, openxrTexture);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, m_swapchain->width(), m_swapchain->height(), 0, GL_RGBA, GL_UNSIGNED_BYTE, nullptr);
+
+    auto& display = m_glContext.display();
+    auto image = display.createEGLImage(m_glContext.platformContext(), EGL_GL_TEXTURE_2D, (EGLClientBuffer)(uint64_t)openxrTexture, { });
```

Comment:
So, we keep a reference to the context just to get the display and platform context, but this is always called with the context made current, so we can avoid keeping the ref and just use GLContext::current()